cgi-bin/LJ/CleanHTML.pm

```diff
@@ -1845,6 +1900,54 @@ sub quote_html {
     return $string;
 }
 
+# The part about which format IDs exist, are aliases, etc. is sort of duplicated
+# info from DW::Formats. But we're keeping the actual formatting behavior here
+# in the cleaner because it seems like a good idea to keep all other code from
+# knowing too much about the cleaner's internals. -NF
+my %markup_formats = (
+    html_casual0 => {
+        formatting  => 'html',
+        addbreaks   => 1,
+        at_mentions => 0,
+    },
+    html_casual1 => {
+        formatting  => 'html',
+        addbreaks   => 1,
+        at_mentions => 1,
+    },
+    html_raw0 => {
+        formatting  => 'html',
+        addbreaks   => 0,
+        at_mentions => 0,
+        noautolinks => 1,
+    },
+    html_extra_raw => {
+        formatting  => 'html',
+        addbreaks   => 0,
+        at_mentions => 0,
+        noautolinks => 1,
+        nodwtags    => 1,
+    },
+    markdown0 => {
+        formatting  => 'markdown',
+        addbreaks   => 0,
+        at_mentions => 1,
+        noautolinks => 1,
+    },
+);
+$markup_formats{markdown} = $markup_formats{markdown0};
+$markup_formats{html}     = $markup_formats{html_casual1};
+
+# Return a group of arguments to pass to clean() based on the requested format
+sub formatting_args {
+    my $format = shift;
```

Comment:
nit: use `$_[0]` for performance reasons (shift does an array modification). Ok, this is a mega-mega-ultra-nit, but I figured I'd mention since it's just a good keyboard habit to have.

Reply:
ah, cool! I never thought that through all the way. ("the more you know" star flies across the background) 

```suggestion
    my $format = $_[0];
```